.circleci/config.yml

```diff
@@ -179,24 +214,22 @@ jobs:
     steps:
       - attach_to_workspace
       - setup_remote_docker
-      - run:
-          name: Build image
-          command: |
-            make docker-otelcol
-            docker tag otelcol:latest otel/opentelemetry-collector:${CIRCLE_TAG:1}
-            docker tag otelcol:latest otel/opentelemetry-collector:latest
-      - run:
-          name: Login to Docker Hub
-          command: docker login -u $DOCKER_HUB_USERNAME -p $DOCKER_HUB_PASSWORD
-      - run:
-          name: Push image
-          command: |
-            docker push otel/opentelemetry-collector:${CIRCLE_TAG:1}
-            docker push otel/opentelemetry-collector:latest
+      - publish_docker_images:
+          repo: opentelemetry-collector
+          tag: ${CIRCLE_TAG:1}
       - run:
           name: Calculate checksums 
-          command: shasum -a 256 bin/* > bin/checksums.txt
+          command: cd bin && shasum -a 256 * > checksums.txt
       - run:
           name: Create Github release and upload artifacts
           command: ghr -t $GITHUB_TOKEN -u $CIRCLE_PROJECT_USERNAME -r $CIRCLE_PROJECT_REPONAME --replace $CIRCLE_TAG bin/
 
+  publish-dev:
```

Comment:
No required jobs?

Reply:
They are listed under the workflows section above.

Reply:
Ok, this is very strange layout for the file - common functions, workflow (depends on jobs) then jobs(depends on common functions). This is what confused me so probably would be good to follow a top to bottom or bottom to top layout.